client/main.lua

```diff
@@ -1,274 +1,216 @@
-            local AmountOfSeats = GetVehicleModelNumberOfSeats(GetEntityModel(Vehicle))
-            for i = 1, AmountOfSeats do
-                local newIndex = #VehicleMenu.items[seatIndex].items+1
-                VehicleMenu.items[seatIndex].items[newIndex] = {
-                    id = i - 2,
-                    title = seatTable[i] or Lang:t("options.other_seats"),
-                    icon = 'caret-up',
-                    type = 'client',
-                    event = 'qb-radialmenu:client:ChangeSeat',
-                    shouldClose = false,
-                }
+    local vehicleitems = {{
+        id = 'vehicle-flip',
+        label = Lang:t("options.flip"),
+        icon = 'car-burst',
+        onSelect = function()
+            TriggerEvent('radialmenu:flipVehicle')
+            lib.hideRadial()
+        end,
+    }}
+
+    vehicleitems[#vehicleitems+1] = convert(Config.VehicleDoors)
+    if Config.EnableExtraMenu then vehicleitems[#vehicleitems+1] = convert(Config.VehicleExtras) end
+
+    if Config.VehicleSeats then
+
+        CreateThread(function()
+            while true do
+                Wait(50)
+                if IsControlJustPressed(0, 23) and not IsPedInAnyVehicle(cache.ped, false) then
+                    local vehicle, _ = QBCore.Functions.GetClosestVehicle(GetEntityCoords(cache.ped))
+                    if vehicle then
+                        local vehicleseats = {}
+                        local seatTable = {
+                            [1] = Lang:t("options.driver_seat"),
+                            [2] = Lang:t("options.passenger_seat"),
+                            [3] = Lang:t("options.rear_left_seat"),
+                            [4] = Lang:t("options.rear_right_seat"),
+                        }
+
+                        local AmountOfSeats = GetVehicleModelNumberOfSeats(GetEntityModel(vehicle))
+                        for i = 1, AmountOfSeats do
+                            vehicleseats[#vehicleseats+1] = {
+                                id = 'vehicleseat'..i,
+                                label = seatTable[i] or Lang:t("options.other_seats"),
+                                icon = 'caret-up',
+                                onSelect = function()
+                                    if IsPedInAnyVehicle(cache.ped, false) then
+                                        TriggerEvent('radialmenu:client:ChangeSeat', i, seatTable[i] or Lang:t("options.other_seats"))
+                                    else
+                                        QBCore.Functions.Notify(Lang:t('error.not_in_vehicle'), 'error')
+                                    end
+                                    lib.hideRadial()
+                                end,
+                            }
+                        end
+                        lib.registerRadial({
+                            id = 'vehicleseatsmenu',
+                            items = vehicleseats
+                        })
+                    end
```

Comment:
Could this be turned into a gate function?

Reply:
I think i did that ? idk

Reply:
You'll be adding an extra layer of abstraction to achieve this (i.e. extracting this work into it's own function).

A simplification of this would appear like:
```lua
local function createVehicleMenu()
    -- do some work
end

CreateThread(function()
    while true do
        Wait(50)
        createVehicleMenu()
    end
end)
```
This makes it clearer to see what the thread is doing and also allows you to sort out the nesting within the loop.

This also will make it easier for someone in the future to attempt to extract a common function if there is any.

This is not necessary though, so I'll leave this to your discretion.